packages/core/assign.js

```diff
@@ -1,21 +1,18 @@
-var getOwnPropertyKeys = require("./get-own-property-keys");
+'use strict';
 
-function assignOne(dst, src) {
+const getOwnPropertyKeys = require('./get-own-property-keys');
+
+const { defineProperty, getOwnPropertyDescriptor } = Reflect;
+
+const assignOne = (dst, src) => {
   if (src != null) {
     // We need to copy regular properties, symbols, getters and setters.
-    var keys = getOwnPropertyKeys(src);
-    for (var i = 0; i < keys.length; i += 1) {
-      var desc = Object.getOwnPropertyDescriptor(src, keys[i]);
-      Object.defineProperty(dst, keys[i], desc);
-    }
+    getOwnPropertyKeys(src).forEach((key) => defineProperty(dst, key, getOwnPropertyDescriptor(src, key)));
```

Comment:
This `assign` is a very core function called almost every time a stamp is created, or an object is created from a stamp. Please, leave the performance optimisations intact. Namely, the `for` loops. We don't want to create extract closures. It impacts memory and CPU usage.

Same for the `reduce` below.

Sorry. I should have left more comments in the code explaining the choice.

Reply:
Now with `for (... of ...)` inside